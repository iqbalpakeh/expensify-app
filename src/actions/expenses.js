import database from "../firebase/firebase";

// -------------------------------------------------
// Before firebase integration:
// 1. Action generator returns object
// 2. Component dispatches object
// 3. Redux store changes
//
// After firebase integration:
// 1. Action generator returns function
// 2. Component dispatches function (?)
// 3. Function runs (has the ability to dispatch other actions and do whatever it wants)
//
// -------------------------------------------------

// -------------------------------------------------
// ACTION ADD_EXPENSE
// -------------------------------------------------

export const addExpense = expense => ({
	type: "ADD_EXPENSE",
	expense
});

export const startAddExpense = (expenseData = {}) => {
	return dispatch => {
		const {
			description = "",
			note = "",
			amount = 0,
			createdAt = 0
		} = expenseData;
		const expense = { description, note, amount, createdAt };
		return database
			.ref("expenses")
			.push(expense)
			.then(ref => {
				dispatch(
					addExpense({
						id: ref.key,
						...expense
					})
				);
			});
	};
};

// -------------------------------------------------
// ACTION REMOVE_EXPENSE
// -------------------------------------------------

export const removeExpense = ({ id } = {}) => ({
	type: "REMOVE_EXPENSE",
	id
});

// -------------------------------------------------
// ACTION EDIT_EXPENSE
// -------------------------------------------------

export const editExpense = (id, updates) => ({
	type: "EDIT_EXPENSE",
	id,
	updates
});
