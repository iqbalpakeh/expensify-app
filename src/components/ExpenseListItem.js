import React from "react";

export default props => (
	<div>
		<p>
			{props.description} {props.amount} {props.createdAt}
		</p>
	</div>
);
