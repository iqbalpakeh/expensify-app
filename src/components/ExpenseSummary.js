import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import getTotalExpense from "../selectors/expenses-total";

export const ExpenseSummary = props => {
	const amount = getTotalExpense(props.expenses);
	const formattedAmount = numeral(amount / 100).format("$0,0.00");
	return (
		<div>
			<h1>
				Viewing {props.expenses.length} totaling {formattedAmount}
			</h1>
		</div>
	);
};

const mapStateToProps = state => ({
	expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseSummary);
