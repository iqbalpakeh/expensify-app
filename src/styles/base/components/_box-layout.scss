.box-layout {
	background: url("/images/bg.jpg");
	background-size: cover;
	height: 100vh;
	width: 100vw;

	display: flex;
	align-items: center;
	justify-content: center;

	&__box {
		background: fade-out(white, 0.15);
		border-radius: 3px;
		padding: $l-size $m-size;
		text-align: center;
		width: 25rem;
	}

	&__title {
		margin: 0 0 $m-size 0;
		line-height: 1;
	}
}
