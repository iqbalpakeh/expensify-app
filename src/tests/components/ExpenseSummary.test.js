import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";
import expenses from "../fixtures/expenses";

test("should render 2 expenses correctly", () => {
	const wrapper = shallow(
		<ExpenseSummary expenses={[expenses[0], expenses[1]]} />
	);
	expect(wrapper).toMatchSnapshot();
});

test("should render 1 expenses correctly", () => {
	const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]} />);
	expect(wrapper).toMatchSnapshot();
});
