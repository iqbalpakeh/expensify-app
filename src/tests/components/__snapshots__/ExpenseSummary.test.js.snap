// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     totaling 
    $1.95
  </h1>
</div>
`;

exports[`should render 2 expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
    2
     totaling 
    $1,096.95
  </h1>
</div>
`;
